### PR TITLE
fix(470): require template validator inputs/outputs object structures

### PR DESCRIPTION
**Pull request by FenrirUnbound**
## Context

By default, keys in a `Joi.object()` are optional. This means that if a key is omitted or missing, it will not be validated against the prescribed rules. 

For instance, if the input was given an empty object, it will accept it as-is. We want the input and output to have a reasonably consistent structure.

## Objective 

This change aims to amend that behavior by making it explicitly known whether the schema keys are optional or required.

## Miscellaneous

* Contributes to screwdriver-cd/screwdriver#480

### CODE REVIEW

File: api/templateValidator.js [hunk 1]
```diff
@@ -6,25 +6,36 @@ const Joi = require('joi');
...
         message: Joi.string()
+            .required()
             .label('Description of a particular validation error'),
         path: Joi.string()
+            .required()
             .label('Dot-notation path to the field that caused the validation error'),
         type: Joi.string()
+            .required()
             .label('The the Joi-type that categorizes the error')
     });
```
Comment (d2lam) [inline api/templateValidator.js]: I think you can use `requiredKeys()` here instead

File: api/templateValidator.js [hunk 2]
```diff
@@ -6,25 +6,36 @@ const Joi = require('joi');
...
     .keys({
-        errors: Joi.array().items(TEMPLATE_ERROR),
+        errors: Joi.array()
+            .items(TEMPLATE_ERROR)
+            .required()
+            .label('Array of errors encountered while validating the given template'),
         // since a template could be parseable but invalid, the contents are unpredicatble
         template: Joi.object()
+            .required()
+            .label('The end-result of parsing the given template')
```
Comment (d2lam) [inline api/templateValidator.js]: same here

File: api/templateValidator.js (hunk 1 above)
Comment (FenrirUnbound) [inline api/templateValidator.js]: Thanks! I didn't know that existed. Makes it sooooo much nicer 😺

File: api/templateValidator.js [hunk 3]
```diff
@@ -6,25 +6,33 @@ const Joi = require('joi');
 const TEMPLATE_ERROR = Joi.object()
     .keys({
         context: Joi.object()
+            .optional()
```
Comment (d2lam) [inline api/templateValidator.js]: you won't need this here anymore I think

**Pull request merged**